Replace USERS & PRIVILEGES with USERMANAGEMENT for snapshot restore (#15164)
crate/crate: docs/appendices/release-notes/5.6.0.rst

Breaking Changes
================

> - When :ref:`restoring a snapshot<sql-restore-snapshot>`, ``USERS`` and

Author: Not sure if we need to also mention something in Deprecations, given that we still support the ``USERS`` and ``PRIVILEGES``, but they restore everything as well.

Reviewer: In general I think we should deprecate the `USERS` and `PRIVILEGES` keywords, such mark them as deprecated and also log a deprecation warning. 
I'd also would add a short entry at the Deprecations section, won't hurt and makes it more explicit.

Author: Should I keep them also in restore-snapshot reference doc: https://github.com/crate/crate/pull/15164/files#diff-8c39d0f3e1b5e04078872d2f44fc36ab109d5ed14b86d9e715001457487ab18cL35
and add a comment as `-- deprecated` ?

Author: Since those are just a key entry in the map here: https://github.com/crate/crate/pull/15164/files#diff-fd5bb986e71a6a438858f7db2f17b7a2c68a3cac25cd673d2ce7c94ee9e0c0e7R56
I don't see how to "mark them deprecated", or you mean something else?

And thx! I will work on a deprecation warning message.

Reviewer: > Should I keep them also in restore-snapshot reference doc: https://github.com/crate/crate/pull/15164/files#diff-8c39d0f3e1b5e04078872d2f44fc36ab109d5ed14b86d9e715001457487ab18cL35 and add a comment as `-- deprecated` ?

Ah yes, as long as they work we should list them I think (but comment as deprecated).

Reviewer: > Since those are just a key entry in the map here: https://github.com/crate/crate/pull/15164/files#diff-fd5bb986e71a6a438858f7db2f17b7a2c68a3cac25cd673d2ce7c94ee9e0c0e7R56 I don't see how to "mark them deprecated", or you mean something else?

Extend the comment to mention not only BWC but deprecated?

Author: sure, thought you have something else with `@Deprecated` annotation in mind.

  ``PRIVILEGES`` keywords, used to restore user management metadata has been
  replaced by ``USERMANAGEMENT``, which dictates that all users and roles of the

>   database together with their privileges are restored. Therefore, now. there is
>   no way to separately restore ``USERS`` or ``PRIVILEGES``, which was allowed
>   before and could lead to several issues.

Reviewer: ```suggestion
  database together with their privileges are restored. Restoring ``USERS`` or
  ``PRIVILEGES`` separately is not possible anymore.
```
